scripts/select-region-for-e2e-test.ts

```diff
@@ -0,0 +1,37 @@
+import crypto from 'crypto';
+import { AWS_REGIONS_TO_RUN_TESTS } from './cci-utils';
+
+/**
+ * This script prints region assignment for an e2e test job.
+ * The algorithm takes input via environment variable - TEST_SUITE
+ * and computes deterministic but random region assignment.
+ * In order to reshuffle regions 'salt' constant below should be modified.
+ *
+ * If region is already assigned, i.e. CLI_REGION environment variable is set this script is pass-through.
+ */
+
+const toNumber = (buf: Buffer) => {
+  return buf.readUInt16BE(0) * 0xffffffff + buf.readUInt32BE(2);
+};
+
+// Algorithm from https://github.com/watson/hash-index/blob/fd5d0606926a821166e428b6e266f03b4dc5d817/index.js#L19
+const computeHashIndex = (input: string, max: number) => {
+  return toNumber(crypto.createHash('sha1').update(input).digest()) % max;
+};
+
+// if region is specified by job honor it.
+let selectedRegion = process.env.CLI_REGION;
+if (!selectedRegion) {
+  const testSuite = process.env.TEST_SUITE;
+  if (!testSuite) {
+    throw Error('TEST_SUITE environment variable must be set');
+  }
+  // See https://en.wikipedia.org/wiki/Salt_(cryptography).
+  // Salt should be changed if we want re-shuffle regions but keep generation deterministic.
+  const salt = 'foo';
+  const regionIndex = computeHashIndex(`${testSuite}${salt}`, AWS_REGIONS_TO_RUN_TESTS.length);
```

Comment:
what kind of distribution do we get, is it an even distribution across the regions?

Reply:
what if we take the job name, and its index in the 'wait_for_ids.json' file, to create an even distribution?

ie,
indexOfJobInJobsList = wait_for_ids.find(..) + randomsalt;
CLI_REGION = REGIONS[indexOfJobInJobsList % REGIONS.length]


similarly, we can shuffle regions by shifting the random salt

code is probably off, but general idea

Reply:
The hashing function as is seems to assign regions pretty evenly.

![image](https://github.com/aws-amplify/amplify-cli/assets/5849952/cdddc88f-59de-45d3-a8b2-b080efbb8dda)

![image](https://github.com/aws-amplify/amplify-cli/assets/5849952/5fd3681c-955c-4fd9-a6e5-9c25e8cb8e67)


The `wait_for_ids.json` is interesting option though. I'll give it a try.